File: 2-2.py
#!/usr/bin/env python

num_valid = 0

f = open('2.input')
for line in f.readlines():
    count, allowed, passwd = line.split(' ')
    pos1, pos2 = count.split('-')
    pos1 = int(pos1) - 1
    pos2 = int(pos2) - 1
    c = allowed[0]
    if (passwd[pos1] == c or passwd[pos2] == c) and not (passwd[pos1] == c and passwd[pos2] == c):
        num_valid +=1

print(num_valid)

File: 2-2.rb
num_valid = 0

File.readlines('2.input').each do |row|
  positions, allowed, pass = row.split(' ')
  pos1 = positions.split('-').first.to_i - 1
  pos2 = positions.split('-').last.to_i - 1
  c = allowed[0]
  if (pass[pos1] == c or pass[pos2] == c) and
      not (pass[pos1] == c and pass[pos2] == c) then
    num_valid += 1
  end
end

print("#{num_valid}\n")

